addons/mail/controllers/main.py

```diff
@@ -131,7 +131,13 @@ def read_followers(self, res_model, res_id):
         request.env['mail.followers'].check_access_rights("read")
         request.env[res_model].check_access_rights("read")
         request.env[res_model].browse(res_id).check_access_rule("read")
-        follower_recs = request.env['mail.followers'].search([('res_model', '=', res_model), ('res_id', '=', res_id)])
+        follower_recs = request.env['mail.followers'].search([
+            ('res_model', '=', res_model),
+            ('res_id', '=', res_id),
+            '|', ('channel_id', '=', False),
+            '|', ('channel_id.public', '!=', 'private'),
+            ('channel_id.channel_partner_ids', 'in', [request.env.user.partner_id.id])
```

Comment:
Note: `!= private` is not equal to `public`, as it is a selection field with multiple values.